README.md

```diff
@@ -416,6 +416,38 @@ server.addHandler( new FileFallbackHandler(_fs, "/jquery/jq1.11.1.js" , "/jquery
 server.addHandler( new FileFallbackHandler(_fs, "/jquery/jqm1.4.5.js" , "/jquery/jqm1.4.5.js" ,   "http://code.jquery.com/mobile/1.4.5/jquery.mobile-1.4.5.min.js",  "max-age=86400"));
 ```
 
+### Serving static files from SPIFFS
+Serve a specific file or attach a folder in SPIFFS to a root url.
+An optional Cache-Control header can be added to responses.
+An optional Last-Modified header can be added, if the request contains If-Modified-Since header with the same value the server will respond with 304 code instead of serving the file.
+```cpp
+
+// Serve static file example
+server.serveStatic("/url/file.htm", SPIFFS, "/dir/file.htm");
+// Serve static folder example
+server.serveStatic("/root-url", SPIFFS, "/root-dir");
+// Specify cache control
+server.serveStatic("/root-url", SPIFFS, "/root-dir", "max-age=86400");
+// Specify last modified (should be updated each time files are updated)
+server.serveStatic("/root-url", SPIFFS, "/root-dir", NULL, "Wed, 08 Jun 2016 22:00:00 GMT");
+
+```
+
+### HTTP Redirect (302)
+Return HTTP 302 response to redirect the client to a different location.
+An optional IPAddress can specify redirection exclusions, for example, exclude redirection to requests to the AP address.
+```cpp
+// Redirect all requests
+server.redirect("/root/file.ext", "/other-location/other-file.ext");
+// Redirect requests not for the AP address
```

Comment:
Indeed, this is same functionality as in FileFallbackHandler but in a more generic design: the redirect is general redirect regardless of file serving. If there is no redirect (for example, call to the AP IP address), the server will fallback to whatever handler can handle the request (this can be static file from SPIFFS or SD card, or even a callback handler. I agree that it is better to avoid redirection altogether and design the solution to perform the request to the appropriate url to begin with.
As for redirecting to CDN - this solution assume that the browser have internet access when request is done on STA mode, however, this might not be true. Another approach might be to serve different static file based on the client local IP, that is, for requests done on AP we serve a file that link to the local resources, for requests done on STA we serve a different file that link to the resources on the CDN. This will remove the need for the ESP to redirect.
Last thing - I agree that it will be good to set the modified date without a change to the sketch, one option is to load the modified date from a config file and update this file when you upload new files to serve, however, this will require a restart to the ESP in order to do the setup again to setup the server with the new date. We need some API to set the modified date for the handler after it was added to avoid restart.